File: 0071-simplify-path/0071-simplify-path.kt
class Solution {
    fun simplifyPath(path: String): String {
        val pathList = path.split(Regex("/+")).filter{ it.isNotEmpty() }
        val stack = Stack<String>()
        for (p in pathList) {
            if (p.equals("..")) {
                if (stack.isNotEmpty()) stack.pop()
            } else if (p.equals(".")) {

            } else {
                stack.push(p)
            }
        }

        return "/" + stack.joinToString(separator = "/")
    }
}

File: 0071-simplify-path/0071-simplify-path.ts
function simplifyPath(path: string): string {

    const pathList = path.split(/\/+/).filter(i => i).filter(i => i != ".");
    let stack = [];
    for (let p of pathList) {
        if (p === "..") {
            stack.pop()
        } else {
            stack.push(p)
        }
    }
    return "/" + stack.join("/")
};